application/plugin_base/item_list.py

```diff
@@ -1655,11 +1673,34 @@ def get_provider(self, path=None):
 		if path is None:
 			return self._current_provider
 
+		# create a local provider if no provider exists yet as fallback in case we are opening an archive
+		if len(self._providers) == 0:
+
+			Provider = self._parent.get_provider_by_protocol('file')
+			result = Provider(self, user.home)
+
+			# cache provider for later use
+			root_path = result.get_root_path(user.home)
+			self._providers[root_path] = result
+			self._current_provider = result
+
 		# check if there is a provider for specified path
 		if path in self._providers:
 			result = self._providers[path]
 
 		else:
+
+			# assuming a path like /home/user/some.zip/subdir1/subdir2
+			# creates intermediate archive provider if necessary
+			p = path
+			while p not in self._providers and p != '/' and '://' not in path:
```

Comment:
Alright - valid concerns - I've just commited/pushed another change but I
will address the two points you made:

1) as I need an existing provider to open the archive file with - let me
try to delay that until it is really needed and
2) I will setup a panel with some ftp://.../file.zip - in this case the
provider from 1) above should not be a LocalProvider but a FtpProvider (in
some generic fashion)

Thanks for your patience :-)


On Sun, Feb 22, 2015 at 12:26 PM, Mladen Mijatov <notifications@github.com>
wrote:

> In application/plugin_base/item_list.py
> <https://github.com/MeanEYE/Sunflower/pull/65#discussion_r25131583>:
>
> >  		# check if there is a provider for specified path
> >  		if path in self._providers:
> >  			result = self._providers[path]
> >
> >  		else:
> > +
> > +			# assuming a path like /home/user/some.zip/subdir1/subdir2
> > +			# creates intermediate archive provider if necessary
> > +			p = path
> > +			while p not in self._providers and p != '/' and '://' not in path:
>
> Creating LocalProvider by default is not good. If path is say on FTP,
> default provider would need to be one for FTP, not LocalProvider.
> Providers should be created as they are needed. Now you point out a problem
> which I overlooked initially, when restoring full path and not entering
> manually level by level.
>
> I asked what that check does because it wasn't initially clear to me but
> now I see that you are actually restricting archive providers only to local
> file system and excluding all URI formed paths. This is not good and we
> should not decide for user when they want to access archive.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MeanEYE/Sunflower/pull/65/files#r25131583>.
>